File: src/main/java/infinitespire/relics/CheckeredPen.java
package infinitespire.relics;

import com.megacrit.cardcrawl.actions.AbstractGameAction;
import com.megacrit.cardcrawl.actions.common.DamageRandomEnemyAction;
import com.megacrit.cardcrawl.actions.utility.UseCardAction;
import com.megacrit.cardcrawl.cards.AbstractCard;
import com.megacrit.cardcrawl.cards.DamageInfo;
import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
import infinitespire.InfiniteSpire;
import infinitespire.abstracts.Relic;

public class CheckeredPen extends Relic {

	public static final String ID = InfiniteSpire.createID("CheckeredPen");

	public CheckeredPen() {
		super(ID, "checkeredpen", RelicTier.UNCOMMON, LandingSound.CLINK);
	}

	@Override
	public void onUseCard(AbstractCard targetCard, UseCardAction useCardAction) {
		AbstractDungeon.actionManager.addToBottom(new DamageRandomEnemyAction(new DamageInfo(null, 1), AbstractGameAction.AttackEffect.BLUNT_LIGHT));
	}
}

File: src/main/kotlin/com/blanktheevil/infinitespire/relics/CheckeredPen.kt
package com.blanktheevil.infinitespire.relics

import com.blanktheevil.infinitespire.extensions.makeID
import com.megacrit.cardcrawl.actions.AbstractGameAction
import com.megacrit.cardcrawl.actions.common.DamageRandomEnemyAction
import com.megacrit.cardcrawl.actions.utility.UseCardAction
import com.megacrit.cardcrawl.cards.AbstractCard
import com.megacrit.cardcrawl.cards.DamageInfo

class CheckeredPen : Relic(ID, IMG, TIER, SOUND) {
  companion object {
    val ID = "CheckeredPen".makeID()
    private const val IMG = "checkeredpen"
    private val TIER = RelicTier.UNCOMMON
    private val SOUND = LandingSound.CLINK
  }

  override fun onUseCard(targetCard: AbstractCard?, useCardAction: UseCardAction?) {
    addToBot(
      DamageRandomEnemyAction(
        DamageInfo(
          null,
          1,
          DamageInfo.DamageType.THORNS
        ),
        AbstractGameAction.AttackEffect.BLUNT_LIGHT
      )
    )
  }
}